src/shared/getComponentOption.js

```diff
@@ -67,3 +62,17 @@ export default function getComponentOption (opts, result = {}) {
   }
   return result
 }
+
+function mergeDataInResult (data, result, component, arrayMerge) {
+  // if option is a function, replace it with it's result
+  if (typeof data === 'function') {
+    data = data.call(component)
+  }
+
+  if (typeof data === 'object') {
+    // merge with existing options
+    return deepmerge(result, data, { arrayMerge })
+  } else {
```

Comment:
When extracting the code into a separate function, you could also get rid of that unnecessary else ☺️

Reply:
Sure! I'll do that as soon as I'm back home 👍